app/uk/gov/hmrc/economiccrimelevyregistration/services/RegistrationValidationService.scala

```diff
@@ -326,4 +337,59 @@ class RegistrationValidationService @Inject() (clock: Clock, schemaValidator: Sc
 
   private def missingErrorMessage(missingDataDescription: String): String = s"$missingDataDescription is missing"
 
+  private def validateOtherEntity(
+    registration: Registration
+  ): ValidationResult[Either[EclSubscription, Registration]] = {
+    val otherEntityJourneyData = registration.otherEntityJourneyData
+    (
+      validateAmlSupervisor(registration),
+      validateOptExists(registration.businessSector, "Business sector"),
+      validateContactDetails("First", registration.contacts.firstContactDetails),
+      validateSecondContactDetails(registration.contacts),
+      validateAmlRegulatedActivity(registration),
+      validateOptExists(registration.relevantAp12Months, "Relevant AP 12 months choice"),
+      validateOptExists(registration.relevantApRevenue, "Relevant AP revenue"),
+      validateConditionalOptExists(
+        registration.relevantApLength,
+        registration.relevantAp12Months.contains(false),
+        "Relevant AP length"
+      ),
+      validateRevenueMeetsThreshold(registration),
+      validateOptExists(registration.optOtherEntityJourneyData, "other entity data"),
+      validateCharity(registration, otherEntityJourneyData)
+    ).mapN {
+      (
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _,
+        _
+      ) => Right(registration)
+    }
+  }
+
+  private def validateCharity(
+    registration: Registration,
+    data: OtherEntityJourneyData
```

Comment:
Why do you need to pass through `data` when it is part of the `registration` you are already passing through?

Reply:
I don't need to.

Reply:
ok may as well remove it then

Reply:
Removed.